### PR TITLE
Check system status after snapper rollback

### PR DESCRIPTION
**Pull request by QingmingSu**
Verify the system is properly rolled back, including:
 1) rollback.service is running
 2) repository version is correct
 3) registration status is synced up

- Related ticket: https://progress.opensuse.org/issues/34069
- Needles: https://gitlab.suse.de/openqa/os-autoinst-needles-sles/merge_requests/798
- Verification run: http://openqa-apac1.suse.de/tests/681#step/snapper_rollback/43

### CODE REVIEW

File: lib/migration.pm [hunk 1]
```diff
@@ -110,5 +111,34 @@ sub record_disk_info {
...
+    script_run("zypper lr -u | tee /dev/$serialdev");
+    my $incorrect_repos = script_output("
+        version=\$(grep VERSION= /etc/os-release | cut -d'=' -f2 | cut -d' ' -f1 | sed 's/\"//g')
+        base_version=\$(echo \$version | cut -d'-' -f1)
+        zypper lr | cut -d'|' -f3 | gawk '/SLE/ || /openSUSE/' | sed \"/\$version\\|Module.*\$base_version/d\"
+    ", 100);
+    record_info('Incorrect Repos', $incorrect_repos, result => 'fail') if $incorrect_repos;
+
+    # Check SUSEConnect status for SLE
+    if (is_sle) {
```
Comment (okurz) [inline lib/migration.pm]: how about an early return from this `sub run`? IMHO lower indendation (== complexity) should be preferred, e.g.:

```perl
return unless is_sle;
…
```

but it's your choice :)

File: lib/migration.pm [hunk 2]
```diff
@@ -110,5 +111,34 @@ sub record_disk_info {
...
+        # check rollback-helper service is enabled and worked properly
+        my $output = script_output "systemctl status rollback.service";
+        if ($output !~ /enabled.*?code=exited,\sstatus=0\/SUCCESS/s) {
+            die "rollback service was failed";
+        }
```
Comment (okurz) [inline lib/migration.pm]: It should be enough to replace the whole block with `systemctl 'is-active rollback';`?

File: lib/migration.pm [hunk 3]
```diff
@@ -110,5 +111,34 @@ sub record_disk_info {
...
+        # system is un-registered during media based upgrade
+        unless (get_var('MEDIA_UPGRADE')) {
```
Comment (okurz) [inline lib/migration.pm]: like above's suggestion: `return if get_var('MEDIA_UPGRADE')` ?

File: lib/migration.pm [hunk 4]
```diff
@@ -110,5 +111,34 @@ sub record_disk_info {
...
+            # Verify registration status matches current system version
+            assert_script_run 'curl -O ' . data_url('console/check_registration_status.py');
+            assert_script_run 'chmod +x check_registration_status.py';
+            assert_script_run './check_registration_status.py';
```
Comment (okurz) [inline lib/migration.pm]: I guess you can simplify these three lines to `assert_script_run '$(curl -s ' . data_url('console/check_registration_status.py') . ')';`

Comment (QingmingSu) [reply lib/migration.pm]: Changed to:
assert_script_run('$(curl -s ' . data_url('console/check_registration_status.py') . ')') unless get_var('MEDIA_UPGRADE');

and testing the updates now.

Comment (QingmingSu) [reply lib/migration.pm]: This doesn't work as expected, i.e. http://openqa-apac1.suse.de/tests/685#step/snapper_rollback/39

**Comment by mitiao** [comment]
LGTM

**Pull request merged**